### ISSUE TITLE
Keyboard input corrupted when scrolling in python on a macintosh

### ISSUE THREAD

**Issue by dj-on-github** [nsf/termbox#78]
I found my termbox based program was quitting when scrolling with the touchpad on a macbook.
I traced this to the program receiving ESCs while scrolling, when it should only be getting arrow up or down events.

I wrote a test case that gets counts the events.
Run it, scroll up a few times, then press X to quit and print the results.
Without a time delay in the loop, this works fine. You only get up arrow events.
I added a time delay in the loop to simulate the delay in updating the view in my program. Then 3 other different events are received, including ESC.

Without delay..
$ ./scrolltest.py 
(1, None, 65516, 0, 0, 0, 0, 0) : 14

With delay..
./scrolltest.py 
(1, u'O', 0, 0, 0, 0, 0, 0) : 1
(1, u'B', 0, 0, 0, 0, 0, 0) : 1
(1, None, 65516, 0, 0, 0, 0, 0) : 104
(1, None, 27, 0, 0, 0, 0, 0) : 1

So something is up with the event handling. It seems to be chopping up the escape sequences into smaller chunks before they are interpreted as key strokes.

scrolltest.py...

#!/usr/bin/env python
# -*- encoding: utf-8 -*-
import sys
import time
import termbox

eventdict = dict()
with termbox.Termbox() as tb:
    finish = False
    while finish==False:
        event = tb.poll_event()
        (type, ch, key, mod, w, h, x, y ) = event
        if type==termbox.EVENT_KEY and ch=='X':
            break
        
        if event in eventdict:
            eventdict[event] += 1
        else:
            eventdict[event] = 1
        time.sleep(0.01)
        
for e in eventdict:
    print e,":",eventdict[e]

**Comment by dj-on-github** [comment]
`#!/usr/bin/env python
# -*- encoding: utf-8 -*-
import sys
import time
import termbox

eventdict = dict()
with termbox.Termbox() as tb:
    finish = False
    while finish==False:
        event = tb.poll_event()
        (type, ch, key, mod, w, h, x, y ) = event
        if type==termbox.EVENT_KEY and ch=='X':
            break
        
        if event in eventdict:
            eventdict[event] += 1
        else:
            eventdict[event] = 1
        time.sleep(0.01)
        
for e in eventdict:
    print e,":",eventdict[e]

`

**Comment by nsf** [comment]
Yeah, sadly it's a known fact on osx: https://github.com/nsf/termbox-go/issues/132. Not sure what to do about it.

**Comment by rationull** [comment]
The linked termbox-go issue was resolved by adding a short wait when an incomplete escape sequence is encountered. The Escape keypress event is only emitted If the wait elapses before any more characters are encountered.
